core/ops_builtin.rs

```diff
@@ -168,6 +169,26 @@ async fn op_read(
   resource.read_return(buf).await.map(|(n, _)| n as u32)
 }
 
+#[op]
+async fn op_read_all(
+  state: Rc<RefCell<OpState>>,
+  rid: ResourceId,
+) -> Result<ZeroCopyBuf, Error> {
+  let resource = state.borrow().resource_table.get_any(rid)?;
+  let (min, maximum) = resource.size_hint();
+  let size = maximum.unwrap_or(min) as usize;
+
+  let mut buffer = Vec::with_capacity(size);
+  loop {
+    let tmp = ZeroCopyBuf::new_temp(vec![0u8; 64 * 1024]);
```

Comment:
I presume the tmp could be moved out of the loop and only allocated once?

Reply:
```
error[E0382]: use of moved value: `tmp`
```